### ISSUE TITLE
NullReferenceException on BulkRead when using null properties in BulkConfig.UpdateByProperties

### ISSUE THREAD

**Issue by Arma-Branca** [borisdj/EFCore.BulkExtensions#602]
Hello,

I'm currently trying to retrieve the rows of my table, using the alternate keys associated in bulk through BulkRead.

My alternate key contains nullable columns, which means that I need to use BulkRead with null properties which currently throws NullReferenceException in 
```
EFCore.BulkExtensions.TableInfo.GetUniquePropertyValues(Object entity, List`1 propertiesNames, Dictionary`2 fastPropertyDict)
```

To better understand, this is an example code that i have, that illustrates the problem:

```
    public class ProductFilterSettings
    {
        public int ProductFilterSettingsId { get; set; }

        public int? ProductId { get; set; }

        public int? ProductEquivalentId { get; set; }

        public string FilterLabelResourceKey { get; set; }
    }
```
```
            var productFilterSettings = new List<ProductFilterSettings>
            {
                new ProductFilterSettings
                {
                    ProductEquivalentId = null,
                    ProductId = 1,
                    FilterLabelResourceKey = "MyKey",
                },
            };

            BulkConfig bulkConfig = new BulkConfig
            {
                UpdateByProperties = new List<string>
                {
                    nameof(ProductFilterSettings.ProductId),
                    nameof(ProductFilterSettings.ProductEquivalentId),
                    nameof(ProductFilterSettings.FilterLabelResourceKey),
                },
                UseTempDB = true,
            };

            await context.BulkReadAsync(productFilterSettings, bulkConfig, cancellationToken: ct);
```

# The problem

After investigating the code more in depth, it looks like the problem is here, in the property.ToString() which doesn't take in account that the property might be null:

```
        public static string GetUniquePropertyValues(object entity, List<string> propertiesNames, Dictionary<string, FastProperty> fastPropertyDict)
        {
            StringBuilder uniqueBuilder = new StringBuilder(1024);
            string delimiter = "_"; // TODO: Consider making it Config-urable
            foreach (var propertyName in propertiesNames)
            {
                var property = fastPropertyDict[propertyName].Get(entity);
                if (property is Array propertyArray)
                {
                    foreach (var element in propertyArray)
                    {
                        uniqueBuilder.Append(element.ToString());
                    }
                }
                else
                {
                    uniqueBuilder.Append(property.ToString());
                }

                uniqueBuilder.Append(delimiter);
            }
            string result = uniqueBuilder.ToString();
            result = result.Substring(0, result.Length - 1); // removes last delimiter
            return result;
        }
```

In my case the null property is ProductEquivalentId, that by being null causes the NullReferenceException with the given stacktrace:
```
   at EFCore.BulkExtensions.TableInfo.GetUniquePropertyValues(Object entity, List`1 propertiesNames, Dictionary`2 fastPropertyDict)
   at EFCore.BulkExtensions.TableInfo.UpdateReadEntities[T](Type type, IList`1 entities, IList`1 existingEntities)
   at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.<ReadAsync>d__8`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.<ReadAsync>d__8`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.<ReadAsync>d__7`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at EFCore.BulkExtensions.SqlBulkOperation.<ReadAsync>d__7`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at EFCore.BulkExtensions.DbContextBulkTransaction.<ExecuteAsync>d__1`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
```

**Comment by borisdj** [comment]
Fixed with v5.3.2

**Comment by snicro** [comment]
> Fixed with v5.3.2

is this fix also ported to v3 variant of the package, for ef core 3?

**Comment by borisdj** [comment]
Nope, v3 is no longer maintained.
